### PR TITLE
[sqlitecpp] do not use stack protection

### LINKED ISSUE #25760: [sqlitecpp] should not use stack protection
sqlitecpp uses stack protection (with gcc) which causes link errors when using it with mingw64

those errors are hard to fix with cmake (-lspp should appear at end of libraries list)

it is just an option to set this of at configure stage

I'm proposing a PR to fix this

### PR DESCRIPTION
**Pull request by reder2000**
change configuration to turn stack protection off

Fixes #25760

### CODE REVIEW

File: ports/sqlitecpp/portfile.cmake [hunk 1]
```diff
@@ -25,6 +25,7 @@ vcpkg_configure_cmake(
         -DSQLITECPP_INTERNAL_SQLITE=OFF
         -DSQLITE_ENABLE_COLUMN_METADATA=OFF
         -DSQLITECPP_INTERNAL_SQLITE=OFF
+        -DSQLITECPP_USE_STACK_PROTECTION=OFF
```
Comment (Cheney-W) [inline ports/sqlitecpp/portfile.cmake:28]:
```suggestion
        ${USE_STACK_PROTECTION}
```
If the link error only occurs with mingw64, Please set a usage condition for this option before vcpkg_configure_cmake.
e.g.
set(USE_STACK_PROTECTION "-DSQLITECPP_USE_STACK_PROTECTION=ON") #(I'm not sure if this option is ON in other triplets, please help confirm)
if(VCPKG_TARGET_IS_MINGW)
    set(USE_STACK_PROTECTION "-DSQLITECPP_USE_STACK_PROTECTION=OFF")
endif()

**Comment by reder2000** [comment]
I took your suggestions as the cmakelists.txt of sqlitecpp mentions that stack protection does not work on (only) mingw64.

File: ports/sqlitecpp/portfile.cmake [hunk 2]
```diff
@@ -15,6 +15,10 @@ vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
 	sqlcipher SQLITE_HAS_CODEC
 )
 
+if(VCPKG_TARGET_IS_MINGW)
+set(USE_STACK_PROTECTION "-DSQLITECPP_USE_STACK_PROTECTION=OFF")
+endif()
```
Comment (Cheney-W) [inline ports/sqlitecpp/portfile.cmake:20]:
```suggestion
set(USE_STACK_PROTECTION "")
if(VCPKG_TARGET_IS_MINGW)
    set(USE_STACK_PROTECTION "-DSQLITECPP_USE_STACK_PROTECTION=OFF")
endif()
```

**Comment by Cheney-W** [comment]
please run following commands when you git commit a new change:
```
.\vcpkg x-add-version sqlitecpp --overwrite-version
git add .
git commit --amend --no-edit
git push
```

**Pull request merged**